tests/std/tests/P0896R4_ranges_alg_uninitialized_move/test.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <memory>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+#ifndef _CONSTEXPR20_DYNALLOC // TRANSITION, P0784
+#define _CONSTEXPR20_DYNALLOC
+#endif
+
+// Validate that uninitialized_move_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::uninitialized_move_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<false>{}, borrowed<false>{})),
+    ranges::uninitialized_move_result<ranges::dangling, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<false>{}, borrowed<true>{})),
+    ranges::uninitialized_move_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<true>{}, borrowed<false>{})),
+    ranges::uninitialized_move_result<int*, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<true>{}, borrowed<true>{})),
+    ranges::uninitialized_move_result<int*, int*>>);
+
+struct int_wrapper {
+    inline static int constructions = 0;
+    inline static int destructions  = 0;
+
+    static constexpr void clear_counts() {
+        if (!is_constant_evaluated()) {
+            constructions = destructions = 0;
+        }
+    }
+
+    static constexpr int magic_throwing_val = 29;
+    int val                                 = 10;
+
+    constexpr int_wrapper() {
+        if (!is_constant_evaluated()) {
+            ++constructions;
+        }
+    }
+    constexpr int_wrapper(int x) : val{x} {
+        if (!is_constant_evaluated()) {
+            ++constructions;
+        }
+    }
+    constexpr int_wrapper(int_wrapper&& that) {
+        if (that.val == magic_throwing_val) {
+            throw magic_throwing_val;
+        }
+
+        val = exchange(that.val, -1);
+        if (!is_constant_evaluated()) {
+            ++constructions;
+        }
+    }
+
+    _CONSTEXPR20_DYNALLOC ~int_wrapper() {
+        if (!is_constant_evaluated()) {
+            ++destructions;
+        }
+    }
+
+    constexpr int_wrapper& operator=(int_wrapper&& that) {
+        if (that.val == magic_throwing_val) {
+            throw magic_throwing_val;
+        }
+        val = exchange(that.val, -1);
+        return *this;
+    }
+
+    auto operator<=>(const int_wrapper&) const = default;
+};
+STATIC_ASSERT(movable<int_wrapper> && !copyable<int_wrapper>);
+
+template <class T, size_t N>
+struct holder {
+    T* ptr = allocator<T>{}.allocate(N);
+
+    holder()              = default;
+    holder(const holder&) = delete;
+    holder& operator=(const holder&) = delete;

```

Comment:
Should implemented and deleted functions be separated by a newline?

Reply:
Yes, but mooted by simplifying this down to a non-`constexpr` wrapper around aligned storage for `N` `T`s.